sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/YamlUtils.java

```diff
@@ -53,19 +53,19 @@ public class YamlUtils {
           .put(Schema.TypeName.BYTES, str -> BaseEncoding.base64().decode(str))
           .build();
 
-  public static Row toBeamRow(@Nullable String yamlString, Schema schema) {
+  public static @Nullable Row toBeamRow(@Nullable String yamlString, Schema schema) {
     return toBeamRow(yamlString, schema, false);
   }
 
-  public static Row toBeamRow(
+  public static @Nullable Row toBeamRow(
       @Nullable String yamlString, Schema schema, boolean convertNamesToCamelCase) {
     if (yamlString == null || yamlString.isEmpty()) {
       List<Field> requiredFields =
           schema.getFields().stream()
               .filter(field -> !field.getType().getNullable())
               .collect(Collectors.toList());
       if (requiredFields.isEmpty()) {
-        return Row.nullRow(schema);
+        return null;
```

Comment:
Why this change? It seems like non-nullable values are preferable. (Similarly below.)

Reply:
reverted back to null row